torchrl/data/replay_buffers/replay_buffers.py

```diff
@@ -600,7 +612,9 @@ def _add(self, data):
         return index
 
     def _extend(self, data: Sequence) -> torch.Tensor:
-        with self._replay_lock, self._write_lock:
+        is_compiling = torch.compiler.is_dynamo_compiling()
```

Comment:
it's a bit annoying but since we want the code to be compatible with torch 2.0, we need to import `is_dynamo_compiling` like this
```
try:
     from torch.compiler import is_dynamo_compiling
except ImportError:
     from torch._dynamo import is_compiling as is_dynamo_compiling
```

Reply:
Done